controllers/weather.go

```diff
@@ -0,0 +1,165 @@
+	resp, err := http.Get(querystring)
+	if err != nil {
+		// handle error
+		log.Println("Error getting Weather from api.openweathermap.org")
+		return wjson, err
+	}
+	if resp.StatusCode != 200 {
+		log.Println("Code Status invalid: ", resp.StatusCode)
+		str := fmt.Sprintf("404, Not found : City: %v - Country: %v", city, country)
+		return wjson, errors.New(str)
+	}
+	// Closing conection
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Print("Error reading result from api.openweathermap.org")
+		return wjson, errors.New("Error reading result from api.openweathermap.org")
+	}
+
+	// Building response
+	errUn := json.Unmarshal(body, &wjson)
+	if errUn != nil {
+		fmt.Println("Error in unmarshall: ", errUn)
+		return wjson, errors.New("Error Unpacking WeatherAPI info")
+	}
+	return wjson, nil
+}//end GetWeatherFromProvider
+
+// Get ...
+// @Title Get 
+// @Description get Weather by id
+// @Success 200 {object} models.Weather
+// @Failure 403 :id is empty
+// @router / [get]
+func (controller *WeatherController) Get() {
+	/**/
+	log.Print("Handle for Get WeatherController Requests")
+	log.Print(time.Now())
+	// Trying to retrieve the params from URL
+	city := controller.GetString("city") // Mexico
+	country := controller.GetString("country") // mx
+	// city := "Mexico" // Mexico
+	// country := "mx" // mx
+
+	// Calling Handler
+	wjson, err := GetWeatherFromProvider(city, country)
+	if err != nil {
+		controller.CustomAbort(404, err.Error())
+	}
+
+	// response := models.FillingResponse(wjson)
+	weatherdb := models.Weather{}
+	weatherdb = models.FillingDBModel(wjson)
+
+	// Insert Weather to DB
+	o := orm.NewOrm()
+	qs := o.QueryTable(weatherdb) // return a QuerySeter
+	qs.Filter("Location", weatherdb.Location)
+
+	//var weather *models.Weather
+	var lastRow models.Weather
+	insertRow := false
+
+	// Just One row
+	err = qs.OrderBy("-Id").One(&lastRow)
+	if err != nil {
+		log.Println(err)
+		// No previous Rows, insert to DB
+		insertRow = true
+	} else {
+		log.Println("Sucess getting weather objects from db")
+		elapsedTime := time.Until(lastRow.RequestedTime.UTC())
+
+		// log.Println(elapsedTime)
+		log.Printf("seconds: %v, LAPSE: %v, %v", elapsedTime.Seconds(), getLapse(), elapsedTime.Seconds() < getLapse())
+		if elapsedTime.Seconds() < getLapse(){
+			log.Println("New row !!")
+			insertRow = true
+		} else {
+			// still not pass enough time to save another row
+			log.Println("You cant insert yet ")
+		}
+	}// End
+	// If requests pass all validation, save in Databsae
+	if insertRow {
+		fmt.Println("Inserting row ...")
+		id, err := o.Insert(&weatherdb)
+		if err == nil {
+			fmt.Println(id)
+		}
+	}
+	controller.Data["json"] = weatherdb
+	controller.ServeJSON()
+}
+
+// GetAll ...
+// @Title GetAll 
+// @Description retrieve all Weather objects
+// @Success 200 {object} models.WeatherDB
+// @Failure 403 :id is empty
+// @router /all [get]
+func (controller *WeatherController) GetAll() {
```

Comment:
@carojaspy it seems out of scope, although it is valuable it adds more work for implementation and maintenance. Think about it.